src/Microsoft.ML.Data/Transforms/ConvertTransform.cs

```diff
@@ -316,7 +316,11 @@ internal static IDataTransform Create(IHostEnvironment env, Arguments args, IDat
                 var item = args.Column[i];
                 var tempResultType = item.ResultType ?? args.ResultType;
                 DataKind kind;
-                KeyRange range = item.KeyRange ?? (item.Range != null ? KeyRange.Parse(item.Range) : null) ?? args.KeyRange ?? (args.Range != null ? KeyRange.Parse(args.Range) : null);
+                KeyRange range = item.KeyRange ??
+                    (item.Range != null ? KeyRange.Parse(item.Range) : null) ??
+                    (item.ResultType == null ?
+                    (args.KeyRange ??
+                    (args.Range != null ? KeyRange.Parse(args.Range) : null)) : null);
                 if (tempResultType == null)
```

Comment:
Hi @yaeldekel thanks for doing this. Since this is the one line in the codebase that actually changed (aside from the tests) I naturally feel like I should understand it. Maybe I'm just tired, but I've been staring at this thing for about five minutes and I feel no closer to understanding what the heck is going on here than when I started. Would it be possible to actually expand it out and explain what is going on here?